float32[] x_traj
float32[] y_traj
float32[] w_traj
